### ISSUE TITLE
Config option to set sync interval

### ISSUE THREAD

**Issue by StarSmasher44** [nextcloud/desktop#2792]
I hope not using the format is fine, as this is more of a suggestion.

**My use case**
I am currently using nextcloud for most of my archival and backup needs for several programs. Most contain database files of active programs that get updated every few seconds. 

**My problem**
As you can probably imagine, constantly uploading said files kind of constantly occupies both source and target Nextcloud server, raising CPU and disk usage all the time.
I haven't yet found a way to slow this down. Nextcloud is on a VM within that server so it's eventually using a lot of cpu for very little.

**My suggestion**
There is a way to set the interval of forced syncs as far as I have seen. I would love the ability to have a similar function but one that allows me to set the interval between file checking (and therefore limit the file checking/uploading tasks to say, every 5 minutes.

I hope it makes sense, and I feel others could potentially make good use of this as well. Either from a performance standpoint or something else.
 
If there are any questions or feedback, I'd be happy to answer them. Likewise I hope this addition is simple enough to warrant implementation :)

**Comment by er-vin** [comment]
It's not quite designed to deal with live data like this I'd say. It's not like there's really an interval right now, it's more that it's reacting to file change notifications from the operating system...

**Comment by StarSmasher44** [comment]
> It's not quite designed to deal with live data like this I'd say. It's not like there's really an interval right now, it's more that it's reacting to file change notifications from the operating system...

Aha, I was under the impression NextCloud would have a function somewhere that would run and check for changes. And that could have been leveraged to either run as it does now, or check after the set interval.

I am aware this isn't the ideal use case for NextCloud, but there can be many instances where you will deal with a constantly changing/updating file, like a log file/folder as an example.

**Comment by er-vin** [comment]
Yes, but then it doesn't update regularly. Logs tend to be smaller and rotated though. It's different from the "big blobs" of a database.

**Comment by StarSmasher44** [comment]
> Yes, but then it doesn't update regularly. Logs tend to be smaller and rotated though. It's different from the "big blobs" of a database.

It still outlays the issue, logfiles while in use will trigger an upload every time 1 line is added, which would translate to a constant flow of checking for changes, uploading, checking.. etc. All I'd like it to space it out by like a few minutes

**Comment by er-vin** [comment]
Yes, I understood, no worries. :-)

Don't expect this soon to be honest, although I can help review patches. ;-)

**Comment by tokun-mobica** [comment]
Hi.
I've prepared some code that allows to set such an interval.
"localSyncInterval" variable can be set in [Nextcloud] section of client configuration file.
Passed value denotes miliseconds, default is 0 if no proper config record is present.
When this interval is > 0, change notifications are gathered untill this interval ends. 
Interval start is triggered by change notification.
I've tested this change on Linux, seems OK.
Waiting for you feedback
